src/promptflow/promptflow/_sdk/_orm/trace.py

```diff
@@ -75,14 +75,36 @@ class LineRun:
     @staticmethod
     def list(
         session_id: typing.Optional[str] = None,
+        runs: typing.Optional[typing.List[str]] = None,
+        experiments: typing.Optional[typing.List[str]] = None,
     ) -> typing.List[typing.List[Span]]:
         with trace_mgmt_db_session() as session:
             stmt: Query = session.query(Span)
             if session_id is not None:
                 stmt = stmt.filter(Span.session_id == session_id)
-            else:
-                # TODO: fully support query
-                raise NotImplementedError
+            if experiments is not None:
+                stmt = stmt.filter(Span.experiment.in_(experiments))
+            if runs is not None:
+                # note that we expect to include the evaluation traces
+                # so we need to filter by both `batch_run_id` and `referenced.batch_run_id`
+                runs_string = ""
+                for run in runs:
+                    runs_string += f"'{run}',"
```

Comment:
we need to validate the inputs, in case of sql injection attack

Reply:
opened a task to track this, will do this in the future PR.